src/code/z_camera.c

```diff
@@ -543,323 +868,7415 @@ s32 Camera_GetWaterBoxBgCamSetting(Camera* camera, f32* waterY) {
     return (camSetting == CAM_SET_NONE) ? -2 : camSetting;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CC938.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CC958.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CC9C0.s")
-
-s32 D_801B9E5C = 0;
-f32 D_801B9E60 = 0.0f;
+void func_800CC938(Camera* camera) {
+    func_800DDFE0(camera);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CCCEC.s")
+/**
+ * Calculates the angle between points `from` and `to`
+ */
+s16 Camera_CalcXZAngle(Vec3f* to, Vec3f* from) {
+    return CAM_DEG_TO_BINANG(RADF_TO_DEGF(func_80086B30(from->x - to->x, from->z - to->z)));
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CD04C.s")
+// BSS
+s16 D_801EDBF0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CD288.s")
+/**
+ * Calculates a pitch adjustment by sampling a position in front of the focal actor and testing the floor height.
+ * Used to detect if the focal actor is near a slope, edge, or cliff.
+ *
+ * @param camera
+ * @param viewYaw The yaw the camera is facing, gives a direction to sample a position
+ * @param shouldInit
+ *
+ * @return pitchOffset resulting pitch adjustment
+ */
+s16 Camera_GetPitchAdjFromFloorHeightDiffs(Camera* camera, s16 viewYaw, s16 shouldInit) {
+    static f32 sFloorYNear;
+    static f32 sFloorYFar;
+    static CameraCollision sFarColChk;
```

Comment:
:monkaBSS:, since bss gives us so much trouble, especially in function static bss, AFAIK we have left every instance of in function static bss as non-matching with a comment `// needs in-function static bss` and avoid importing the bss.

Reply:
Surprisingly, the in-function bss hasn't caused too much issues for this particular function, but I'll go ahead and make it non-matching

Reply:
There isn't any issues with matching either when it's included, so I'm fine not ifdef'ing

Reply:
In that case it should be fine to not ifdef, we can always undo it if it starts to cause troubles